source/dagger/gameplay/team_game/team_game_main.cpp

```diff
@@ -144,104 +146,109 @@ void SetupWorldSmiljana(Engine& engine_, Registry& reg_) {
         AnimatorPlay(enemyAnimator, "among_them_animations:goblin_idle");
 
-        auto& enemyTransform = reg.emplace<Transform>(enemy);
-        enemyTransform.position = { 0, 25, zPos };
+        auto& enemyTransform = reg_.emplace<Transform>(enemy);
+        enemyTransform.position = { 0, 25, 1 };
 
-        auto& enemyInput = reg.emplace<InputEnemiesFile>(enemy);
+        auto& enemyInput = reg_.emplace<InputEnemiesFile>(enemy);
         enemyInput.pathname = "path.txt";
         enemyInput.currentshape = "goblin";
 
-        reg.emplace<EnemyDescription>(enemy);
+        reg_.emplace<EnemyDescription>(enemy);
 
-        reg.emplace<MovableBody>(enemy);
+        reg_.emplace<MovableBody>(enemy);
 
         //ENEMY NO.2
 
         //ENEMY 
-        auto enemy2 = reg.create();
+        auto enemy2 = reg_.create();
 
         auto& enemy2State = ATTACH_TO_FSM(EnemyFSM, enemy2);
         enemy2State.currentState = EEnemyState::Patrolling;
 
-        auto& enemy2Sprite = reg.emplace<Sprite>(enemy2);
+        auto& enemy2Sprite = reg_.emplace<Sprite>(enemy2);
         AssignSprite(enemy2Sprite, "spritesheets:among_them_spritesheet:bat_anim:1");
         enemy2Sprite.scale = { 1, 1 };
 
-        auto& enemy2Animator = reg.emplace<Animator>(enemy2);
+        auto& enemy2Animator = reg_.emplace<Animator>(enemy2);
         AnimatorPlay(enemy2Animator, "among_them_animations:bat");
 
-        auto& enemy2Transform = reg.emplace<Transform>(enemy2);
-        enemy2Transform.position = { 0, 60, zPos };
+        auto& enemy2Transform = reg_.emplace<Transform>(enemy2);
+        enemy2Transform.position = { 0, 60, 1 };
 
-        auto& enemy2Input = reg.emplace<InputEnemiesFile>(enemy2);
+        auto& enemy2Input = reg_.emplace<InputEnemiesFile>(enemy2);
         enemy2Input.pathname = "pathbat.txt";
         enemy2Input.currentshape = "bat";
 
-        reg.emplace<EnemyDescription>(enemy2);
+        reg_.emplace<EnemyDescription>(enemy2);
 
-        reg.emplace<MovableBody>(enemy2);
+        reg_.emplace<MovableBody>(enemy2);
 
 
 
-    }
 }
+
 void team_game::SetupWorld(Engine &engine_)
 {
     auto& reg = engine_.Registry();
 
-    float zPos = 1.f;
+    auto gameCtrl = reg.create();
+
+    auto& ctrlInput = reg.emplace<InputReceiver>(gameCtrl);
+    ctrlInput.contexts.push_back("AmongThemReload");
+
+    reg.emplace<GameController>(gameCtrl);
 
     // STATIC BODIES MAP
     auto mapEnt = reg.create();
-	  auto& map = reg.emplace<StaticBodyMap>(mapEnt);
-	  Engine::PutDefaultResource<StaticBodyMap>(&map);
+	auto& map = reg.emplace<StaticBodyMap>(mapEnt);
+	Engine::PutDefaultResource<StaticBodyMap>(&map);
 	
     // You can add your own WorldSetup functions when testing, call them here and comment out mine
-    //SetupWorldJovica(engine_, reg);
+    SetupWorldJovica(engine_, reg);
     //SetupWorldKosta(engine_, reg);
-    SetupWorldSmiljana(engine_);
+    //SetupWorldSmiljana(engine_, reg);
 }
```

Comment:
Try putting here that id as a global parameter and set it to 0 outside of SetupWorld and then try something like 
if(id==0) {
SetupWorldJovica(engine_, reg); }
if(id==1) {  
SetupWorldSmiljana(engine_, reg); }
id++;
if(id==2) {id=0;}